src/main/java/net/goldtreeservers/worldguardextraflags/fawe/FaweWorldEditFlagMaskManager.java

```diff
@@ -1,93 +1,111 @@
 	{
 		super(plugin.getName());
-		
 		this.plugin = plugin;
 	}
-	
-    public ProtectedRegion getRegion(Player player, Location loc)
-    {
-        final com.sk89q.worldguard.LocalPlayer localplayer = this.plugin.getWorldGuardCommunicator().wrapPlayer(player);
-        RegionManagerWrapper manager = this.plugin.getWorldGuardCommunicator().getRegionContainer().get(player.getWorld());
-        final ProtectedRegion global = manager.getRegion("__global__");
-        if (global != null && !isDenied(localplayer, global))
-        {
-            return global;
-        }
-        
-        final ApplicableRegionSet regions = manager.getApplicableRegions(player.getLocation());
-        for (final ProtectedRegion region : regions)
-        {
-            if (!isDenied(localplayer, region))
-            {
-                return region;
-            }
-        }
-        return null;
-    }
-
-    public boolean isDenied(LocalPlayer localplayer, ProtectedRegion region)
-    {
-        return region.getFlag(Flags.WORLDEDIT) == State.DENY;
-    }
 
 	@Override
-	public FaweMask getMask(FawePlayer<Player> fawePlayer)
+	public RegionFilter getFilter(String world)
+	{
+		return new WorldGuardFilter(plugin, Bukkit.getWorld(world));
+	}
+
+	@Override
+	public FaweMask getMask(FawePlayer<Player> fp, MaskType type)
 	{
-		return null; //Problems here due to FaweMask using LocalWorld
-		
-		/*final Player player = fawePlayer.parent;
-		final Location location = player.getLocation();
-        final ProtectedRegion myregion = this.getRegion(player, location);
-        
-        if (myregion != null)
-        {
-            final BlockVector pos1;
-            final BlockVector pos2;
-            if (myregion.getId().equals("__global__"))
-            {
-                pos1 = new BlockVector(Integer.MIN_VALUE, 0, Integer.MIN_VALUE);
-                pos2 = new BlockVector(Integer.MAX_VALUE, 255, Integer.MAX_VALUE);
-            }
-            else
-            {
-                pos1 = new BlockVector(myregion.getMinimumPoint().getBlockX(), myregion.getMinimumPoint().getBlockY(), myregion.getMinimumPoint().getBlockZ());
-                pos2 = new BlockVector(myregion.getMaximumPoint().getBlockX(), myregion.getMaximumPoint().getBlockY(), myregion.getMaximumPoint().getBlockZ());
-            }
-            
-            return new FaweMask(pos1, pos2)
-            {
-                @Override
-                public String getName()
-                {
-                    return myregion.getId();
-                }
-            };
-        }
-        else
-        {
-            return null;
-        }*/
+		final Player player = fp.parent;
+		final LocalPlayer localplayer = this.plugin.getWorldGuardCommunicator().wrapPlayer(player);
+		RegionManagerWrapper manager = this.plugin.getWorldGuardCommunicator().getRegionContainer().get(player.getWorld());
+
+		return new FaweMask(new MultiRegion(manager, localplayer), null)
```

Comment:
U can get rid of the anonymous method

Reply:
I'm not too sure if I can (assuming I understand you correctly). The FaweMask needs to be returned as as we don't really use that functionality it needs to override isValid. Unless you would prefer me extend the class separately and return that?